### ISSUE TITLE
Water mask always uses texture coordinate 0, which is often (incorrectly) Web Mercator

### ISSUE THREAD

**Issue by kring** [CesiumGS/cesium-unreal#632]
MF_CesiumWaterMaskSample uses a parameter named `geographicTextureCoordinateIndex`. It's never set, so it will always have its default value (0). If the tileset has a Web Mercator overlay (e.g. Bing), then texture coordinate 0 will use a Web Mercator projection instead of the expected Geographic projection.

This doesn't cause an immediate problem because we fade out the water effect as we zoom out. Because the texture coordinates are effectively correct at the tile edges anyway, and because the difference between Web Mercator and Geographic _within_ a tile is not noticeable except within large tiles, it's nearly impossible to notice any actual effect from this bug. But that might not be true with a custom material that doesn't fade out.

**Comment by kring** [comment]
A simple but hacky solution to this is to add the geographic projection for the water mask before adding any other projections for overlays (currently we do the opposite, for not real reason). That way, when a water mask is present, the geographic coordinates will always be found at index 0.

A better solution is to use the coordinates designated for the geographic projection, no matter what index they're found at. That will require passing a bit of extra watermask-related data from cesium-native to cesium-unreal in extras or whatever.

**Comment by nithinp7** [comment]
Could this be helped by working the water mask into the raster overlay system? Or did we decide not to do that for other reasons?

**Comment by kring** [comment]
Yeah, the water mask should become a raster overlay for sure. There's no good reason it's not already done other than we didn't get to it yet.
